src/test/scala/mesosphere/marathon/tasks/PortWithRoleRandomPortsFromRangesTest.scala

```diff
@@ -63,25 +63,46 @@ class PortWithRoleRandomPortsFromRangesTest extends MarathonSpec {
     )
   }
 
-  test("is lazy with one range") {
-    // otherwise this will be really really slow
+  // benchmark duration to allow calculate relative acceptable times
+  private[this] lazy val benchmarkDuration = {
+    System.gc()
     val start = System.nanoTime()
+    assert((1 to 100000).map(_.toLong).sum == 5000050000L)
+    FiniteDuration(System.nanoTime() - start, TimeUnit.NANOSECONDS)
+  }
 
-    val ports = PortWithRole.lazyRandomPortsFromRanges()(Seq(portRange("role", 1, Integer.MAX_VALUE)))
-    ports.take(3).toSeq
+  test("is lazy with one range") {
+    // if it is not implemented lazily, this will be really really slow
+
+    def performTest(): Unit = {
+      val ports = PortWithRole.lazyRandomPortsFromRanges()(Seq(portRange("role", 1, Integer.MAX_VALUE)))
+      assert(ports.take(3).toSet.size == 3)
+    }
 
+    // warm up
+    (1 to 100).foreach(_ => performTest())
+
+    // real test
+    System.gc()
+    val start = System.nanoTime()
+    performTest()
     val duration = FiniteDuration(System.nanoTime() - start, TimeUnit.NANOSECONDS)
+
+    val allowTimeForTest = benchmarkDuration
```

Comment:
I assume you wanted to use this value in the assert on line 97, correct? Then please go ahead and do so :)